contracts/task-queue/src/contract.rs

```diff
@@ -268,6 +274,62 @@ mod execute {
 
         Ok(res)
     }
+
+    pub fn add_hook(
+        deps: DepsMut,
+        info: MessageInfo,
+        hook_type: TaskHookType,
+        receiver: String,
+    ) -> Result<Response, ContractError> {
+        // Validate the address
+        let receiver = deps.api.addr_validate(&receiver)?;
+
+        let config = CONFIG.load(deps.storage)?;
+
+        // Only allow the hook admin to register hooks
+        if config.hook_admin.map_or(true, |admin| admin != info.sender) {
+            return Err(ContractError::Unauthorized {});
+        }
```

Comment:
this seems to prevent adding a hook if no `hook_admin` is set. is that intentional? or should an empty `hook_admin` allow anyone to add/remove hooks? maybe it should not be optional and instead default to the instantiator?

it makes sense to me that only the AVS manager so to speak can add hooks, or else we'll run into DOS/gas issues. and it's likely undesirable to allow a permanent removal of the hook admin since it may become difficult to swap out a task queue for another

Reply:
Yes, the task hooks were implemented as "optional" functionality, but it would also make sense to just default to the sender. In that case, we should probably just use cw-ownable owner here for any AVS-manager auth including these hooks.